nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -2389,6 +2413,53 @@ mod tests {
         logctx.cleanup_successful();
     }
 
+    #[tokio::test]
+    async fn test_instance_deletion_is_idempotent() {
```

Comment:
nit, take it or leave it: perhaps worth a comment explaining what's being tested here and why it's important?

Reply:
Sounds good, updated!